gitops/snapshot.go

```diff
@@ -152,6 +155,16 @@ func MarkSnapshotIntegrationStatusAsInProgress(adapterClient client.Client, ctx
 	if err != nil {
 		return nil, err
 	}
+	if helpers.HasLabel(snapshot, tekton.PipelineRunEnterIntegrationSvcTimeLabel) {
+		snapshotInProgressTime := &metav1.Time{Time: time.Now()}
+		buildPipelineRunEnteringTime := snapshot.Labels[tekton.PipelineRunEnterIntegrationSvcTimeLabel]
+		buildPipelineRunEnteringIntegrationSvcTimeInt, _ := strconv.ParseInt(buildPipelineRunEnteringTime, 10, 64)
+		buildPipelineRunEnteringIntegrationSvcTime := time.Unix(buildPipelineRunEnteringIntegrationSvcTimeInt, 0)
+		buildPipelineRunEnteringIntegrationSvcMetaTime := &metav1.Time{Time: buildPipelineRunEnteringIntegrationSvcTime}
+
+		go metrics.RegisterRunningSnapshot(buildPipelineRunEnteringIntegrationSvcMetaTime, snapshotInProgressTime)
```

Comment:
I've been thinking about this and based on the requirements of HACBS-1580, I think that we should be measuring the time between the `Snapshot` creation and it being marked as `In Progress` - this would then also cover manually created snapshots.
We could then, instead of deriving the time from the `PipelineRunEnterIntegrationSvcTimeLabel` label, use the `snapshot.GetCreationTimestamp()` as the starting time here.

Reply:
OK. Updated. Please help to review again.

Reply:
Thanks for updating it!